src/main/java/backend/model/event/TeamServiceImpl.kt

```diff
@@ -168,66 +164,46 @@ class TeamServiceImpl : TeamService {
     }
 
     fun getSponsoringSum(team: Team): BigDecimal {
-        val distanceKm = this.getLinearDistanceForTeam(team.id!!)
-        var sponsorSum = BigDecimal.ZERO
-
-        val sponsorings = team.sponsoring.filter { it.status == ACCEPTED || it.status == PAYED }
-
-        sponsorings.forEach { sponsoring ->
-            val amount = sponsoring.amountPerKm.numberStripped.multiply(BigDecimal.valueOf(distanceKm))
-            if (amount.compareTo(sponsoring.limit.numberStripped) == 1) {
-                sponsorSum = sponsorSum.add(sponsoring.limit.numberStripped)
-            } else {
-                sponsorSum = sponsorSum.add(amount)
-            }
-        }
-
-        sponsorSum = sponsorSum.setScale(2, BigDecimal.ROUND_HALF_UP)
-
-        return sponsorSum
-    }
-
-    fun getChallengeSum(team: Team): Map<String, BigDecimal> {
-        var withProofSum = BigDecimal.ZERO
-        var acceptedProofSum = BigDecimal.ZERO
-
-        team.challenges.forEach { challenge ->
-            if (challenge.status == ChallengeStatus.WITH_PROOF) {
-                withProofSum = withProofSum.add(challenge.amount.numberStripped)
+        val distance = this.getLinearDistanceForTeam(team.id!!)
+
+        return team.sponsoring.parallelStream().filter { it.status == ACCEPTED || it.status == PAYED }.map { sponsoring ->
+            val amount = sponsoring.amountPerKm.numberStripped * BigDecimal.valueOf(distance)
+            return@map when (amount > sponsoring.limit.numberStripped) {
+                true -> sponsoring.limit.numberStripped
+                false -> amount
+            }.setScale(2, BigDecimal.ROUND_HALF_UP)
+        }.reduce { acc: BigDecimal, sponsorSum: BigDecimal ->
+            acc + sponsorSum
+        }.orElseGet { BigDecimal.ZERO }
+    }
+
+    fun getChallengeSum(team: Team): ChallengeDonateSums {
+        return team.challenges.parallelStream().map { challenge ->
+            return@map when (challenge.status) {
+                ChallengeStatus.WITH_PROOF -> ChallengeDonateSums(challenge.amount.numberStripped, BigDecimal.ZERO)
+                ChallengeStatus.PROOF_ACCEPTED -> ChallengeDonateSums(BigDecimal.ZERO, challenge.amount.numberStripped)
+                else -> ChallengeDonateSums(BigDecimal.ZERO, BigDecimal.ZERO)
```

Comment:
Die Logik würde ich bei Gelegenheit noch in das Model verschieben, dann kann man damit später auch die Spendensumme und so berechnen